docs/tutorials/ipfs-dir-upload/index.md

```diff
@@ -424,4 +424,48 @@ When you upload a bunch of files, you might notice that this creates some issues
     </div>`
 ```
 
+## Show the user what is happening
+
+Now that everything is working, let's add a modal indicate that the files are being uploaded. We will use another custom event to indicate when an upload starts. Add the following to the `onSubmit` handler in `upload.js` after assigning the `fileInput` variable and before calling `batchUpload`:
+
+```js
+         const fileInput = document.querySelector('#dirForm input[type="file"]')
+         this.dispatchEvent(new CustomEvent('dirUploadStart', { detail: { fileCount: fileInput.files.length } }))        // <--- Add this line
+         const newCid = await batchUpload(fileInput.files, pathPrefix)
+
+```
+
+In `lib.js`, we will handle this new event and display a modal when the upload starts. When the upload completed, we will show a success message and redirect the page after a delay of 5 seconds. Update the code at the end of `loadSidebar` as follows:
+
+```js
+    dirUploadForm.addEventListener('dirUploadStart', e => {
+        const modalDiv = document.createElement('div')
```

Comment:
Would it make sense to use the built in `<dialog>` element instead? https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog

Reply:
Cool, I've made the change.